File: tests/ts/iso_a.js
// iso_a.js — half of the module-isolation pair: same module-scope
// binding names and same exported hook names as iso_b.js. ES-module
// scoping must keep the two fully apart (the lua-_ENV / ruby-harvest
// equivalent, provided by the engine itself).

let e = null;
let n = 0;

export function update(dt) {
  if (e === null) e = Entity.create();
  n += 1;
  e.set("IsoA", { n });
}

export function deinit() {
  labelle.log("iso_a deinit");
}

File: tests/ruby/iso_a.rb
# iso_a.rb — half of the top-level hook isolation pair (with iso_b.rb):
# both scripts define the SAME hook names with their own @ivar state.
# mruby has no lua-style per-chunk _ENV — without the harvest protocol,
# the second definition would clobber the first on Object and one script
# would tick twice while the other went silent.

def init
  @e = Labelle::Entity.create
  @e.set("IsoA", n: 0)
end

def update(dt)
  c = @e.get("IsoA")
  c[:n] += 1
  @e.set("IsoA", c)
end

def deinit
  Labelle.log("iso_a deinit")
end
